experimental/errorsource/error_source_middleware.go

```diff
@@ -11,6 +11,7 @@ import (
 // Error captures error source and implements the error interface
 type Error struct {
 	Source backend.ErrorSource
+	Status backend.Status
```

Comment:
Can/should a plugin developer set this field? 
The constructor, `SourceError`, does not accept a `Status` field., so the only way is setting the field manually:

```go
e := SourceError(...)
e.Status = backend.StatusBadGateway
```
Maybe we can add a new constructor using the options pattern?

```go
errorsource.NewError(
	err,
	backend.ErrorSourceDownstream,
	errorsource.WithStatus(backend.StatusBadGateway),
)
```

Reply:
updated